legacy/src/main/java/org/opensearch/sql/legacy/plugin/RestSQLQueryAction.java

```diff
@@ -101,9 +103,10 @@ public RestChannelConsumer prepareRequest(SQLQueryRequest request, NodeClient no
     try {
       // For now analyzing and planning stage may throw syntax exception as well
       // which hints the fallback to legacy code is necessary here.
-      plan = sqlService.plan(
-                sqlService.analyze(
-                    sqlService.parse(request.getQuery())));
+      LogicalPlan logicalPlan = sqlService.analyze(sqlService.parse(request.getQuery()));
```

Comment:
In the future we may want to also carry planContext when generating the logical plan. Example use case: ML command can set the plan context to specify it needs pagination. Then when building the physical plan, the Planner knows to use scroll instead of regular query.